include/aspect/postprocess/visualization/boundary_velocity_residual.h

```diff
@@ -0,0 +1,144 @@
+       *
+       * The member functions are all implementations of those declared in the
+       * base class. See there for their meaning.
+       */
+      template <int dim>
+      class BoundaryVelocityResidual
+        : public DataPostprocessorVector<dim>,
+		  public SimulatorAccess<dim>,
+          public Interface<dim>
+      {
+        public:
+    	  BoundaryVelocityResidual ();
+
+          void initialize () override;
+
+          void
+          evaluate_vector_field(const DataPostprocessorInputs::Vector<dim> &input_data,
+                                std::vector<Vector<double> > &computed_quantities) const override;
+
+          /**
+           * Declare the parameters this class takes through input files.
+           */
+          static
+		  void
+		  declare_parameters (ParameterHandler &prm);
+
+          /**
+           * Read the parameters this class declares from the parameter file.
+           */
+          void
+		  parse_parameters (ParameterHandler &prm) override;
+
+        private:
+          bool use_spherical_unit_vectors;
+
+          bool use_gps_data;
+
+          /**
+		   * Pointer to the gplates boundary velocity model
+		   */
+          std::unique_ptr<BoundaryVelocity::internal::GPlatesLookup <dim> >  gplates_lookup;
+
+          /**
+		   * Pointer to the  ascii boundary velocity model
+		   */
+          std::unique_ptr<Utilities::AsciiDataLookup<dim>> ascii_data_lookup;
+
+          /**
+           * Directory in which the data files are present.
+           */
+          std::string ascii_data_directory;
+
+          /**
+           * Filename of data file. The file names can contain the specifiers %s
+           * and/or %c (in this order), meaning the name of the boundary and the
+           * number of the data file time step.
+           */
+          std::string ascii_data_file_name;
+
+          /**
+           * Scale the data by a scalar factor. Can be used to transform the
+           * unit of the data (if they are not specified in SI units (m/s or
+           * m/yr depending on the "Use years in output instead of seconds"
+           * parameter).
+           */
+          double ascii_scale_factor;
+
+          /**
+           * Directory in which the data files are present.
+           */
+          std::string gplates_data_directory;
+
+          /**
+           * Filename of data file. The file names can contain the specifiers %s
+           * and/or %c (in this order), meaning the name of the boundary and the
+           * number of the data file time step.
+           */
+          std::string gplates_data_file_name;
+
+          /**
+           * Scale the data by a scalar factor. Can be used to transform the
+           * unit of the data (if they are not specified in SI units (m/s or
+           * m/yr depending on the "Use years in output instead of seconds"
+           * parameter).
+           */
+          double gplates_scale_factor;
```

Comment:
Can we unify these parameters? I.e. could we just have `data_directory`, `data_file_name`, and `scale_factor`? Then the single parameter "Use ascii data" would determine if we load a gplates or an ascii data file.